### ISSUE TITLE
Fix 'test_windows/test_cpe_indexing' tests in 4.2

### ISSUE THREAD

**Issue by damarisg** [wazuh/wazuh-qa#1565]
### Issue information

| Related issue |
|--|
| Closes https://github.com/wazuh/wazuh-qa/issues/1531|

When it comes to starting with the test fix, there is some information that can help you:
```
Module: Vulnerability Detector
Support Team:  "binary beasts"
Target: Manager   
OS: Linux
```
I add some information obtained while researched the logs that failed.

## Case:

|  Type | Description  |
|--|--|
|Test Path | test_windows/test_cpe_indexing
| Consistent | no
| Test Execution | 2/3 executions failed
| Cases Fails | 3
| Summary | On `test_window_version_indexing` when `make_vuln_callback` is called. it requires research to know if there is a problem with parser error or if we don't detect log.
|Error message | "{mock_system['os_name']} was not indexed"

In order to finish this issue the following tasks should be fulfilled:
- [ ] Research of fails.
- [ ] Apply Fix 
- [ ] Full Green/ Full Yellow  in `test_windows/test_cpe_indexing` for 3 times. 
- [ ] Documentation of any important change done for these tests or the used tools.

**Comment by mdengra** [comment]
### 2021-07-16

Test results with the modules `sca`, `syscollector` and `rootcheck` modules disabled in the `ossec.conf`: 

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_cpe_indexing_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6828943/test_cpe_indexing_local_dm_r1.log)| 2021-07-16 | Miguel | :red_circle: 
|[test_cpe_indexing_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6828945/test_cpe_indexing_local_dm_r2.log)| 2021-07-16 | Miguel | :red_circle: 
|[test_cpe_indexing_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6828947/test_cpe_indexing_local_dm_r3.log)| 2021-07-16 | Miguel | :red_circle:

**Comment by mdengra** [comment]
The test keeps failing to increase the timeout from **50** seconds to **120** seconds. You can also confirm in the attached files that the required logs are not written:

| Test result | log | 
|--|--|
|[test_cpe_indexing_local_dm_120s.log](https://github.com/wazuh/wazuh-qa/files/6829388/test_cpe_indexing_local_dm_120s.log)| [logs_test_cpe_indexing_local_dm_120s.txt](https://github.com/wazuh/wazuh-qa/files/6829390/logs_test_cpe_indexing_local_dm_120s.txt)

**Comment by mdengra** [comment]
It seems that the origin of the failure is that the `NVD_CVE` table of the `CVEs` database is empty at the moment of the failure. This is because no vulnerability is explicitly inserted in this table, but rather `Vulnerability Detector` updates it with the local vulnerabilities when it scans the agent. 

To generate the message required by the test, one of the requirements is that the table mentioned above contains data.

The solution is to insert a vulnerability at the beginning of each use case in the `NVD_CVE` table to avoid the waiting time caused by the scanning of the agent by `Vulnerability Detector`.

**Comment by damarisg** [comment]
Test Results:

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[YellowCPEIndexing.log](https://github.com/wazuh/wazuh-qa/files/6900726/YellowCPEIndexing.log)| 2021-07-29| Seyla| 🟡 
|[YellowCPEIndexing2.log](https://github.com/wazuh/wazuh-qa/files/6900714/YellowCPEIndexing2.log)| 2021-07-29| Seyla| 🟡 
|[YellowCPEIndexing3.log](https://github.com/wazuh/wazuh-qa/files/6900728/YellowCPEIndexing3.log)| 2021-07-29| Seyla| 🟡 


This issue is fixed and closed by #1656.

**Comment by mdengra** [comment]
### 2021-08-17

To attempt to reproduce the error that occurs rarely and randomly, tests have been performed with different states of the VM (clean and dirty environment) and also reducing the CPU usage limit to 10% to see if the reduction of resources affected the number of times that the error occurs:

Used Wazuh-QA branch: `1531-full-yellow-vuln-det`
Test results with the default configuration modules in the `ossec.conf`: 

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_cpe_indexing_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6999285/test_cpe_indexing_local_r1.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6999286/test_cpe_indexing_local_r2.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6999288/test_cpe_indexing_local_r3.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_r4.log](https://github.com/wazuh/wazuh-qa/files/6999290/test_cpe_indexing_local_r4.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_cpu10_r1.log](https://github.com/wazuh/wazuh-qa/files/6999316/test_cpe_indexing_local_cpu10_r1.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_cpu10_r2.log](https://github.com/wazuh/wazuh-qa/files/6999319/test_cpe_indexing_local_cpu10_r2.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_cpu10_r3.log](https://github.com/wazuh/wazuh-qa/files/6999323/test_cpe_indexing_local_cpu10_r3.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_cpu10_r4.log](https://github.com/wazuh/wazuh-qa/files/6999325/test_cpe_indexing_local_cpu10_r4.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_cpu10_r5.log](https://github.com/wazuh/wazuh-qa/files/6999326/test_cpe_indexing_local_cpu10_r5.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_cpu10_r6.log](https://github.com/wazuh/wazuh-qa/files/6999329/test_cpe_indexing_local_cpu10_r6.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_cpu10_r7.log](https://github.com/wazuh/wazuh-qa/files/6999331/test_cpe_indexing_local_cpu10_r7.log)| 2021-08-17 | Miguel | :yellow_circle: 
|[test_cpe_indexing_local_cpu10_r8.log](https://github.com/wazuh/wazuh-qa/files/6999336/test_cpe_indexing_local_cpu10_r8.log)| 2021-08-17 | Miguel | :red_circle: 

As you can see, the last run has failed:
```
            with connect:
                for item in query_list:
>                   connect.execute(item)
E                   sqlite3.OperationalError: database is locked
```
This is probably due to the issue: https://github.com/wazuh/wazuh/issues/9309, so this bug should be fixed by that issue.

**Comment by damarisg** [comment]
This issue is closed because the test was disabled by #1602.
